### PR TITLE
BXMSDOC-8108 updated with changes to  Quarkus community doc

### PR DESCRIPTION
**Pull request by emmurphy1**
[BXMSDOC-8108 Update with changes to Quarkus community doc](https://issues.redhat.com/browse/BXMSDOC-8108)

### CODE REVIEW

File: doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,63 @@
+
+[id='business-optimizer-test-constraints-proc_{CONTEXT}']
+= Test the school timetable constraints
+
+To test each constraint of the timetable project in isolation, use a `ConstraintVerifier` in unit tests. Doing this tests each constraint’s corner cases in isolation from the other tests, which lowers maintenance when adding a new constraint with proper test coverage.
```
Comment (briandooley) [inline doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc]:
```suggestion
To test each constraint of the timetable project in isolation, use a `ConstraintVerifier` in unit tests. This tests each constraint’s corner cases in isolation from the other tests, which lowers maintenance when adding a new constraint with proper test coverage.
```

File: doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+This test verifies that the constraint `TimeTableConstraintProvider::roomConflict`, when given three lessons in the same room and two of the lessons have the same timeslot, penalizes with a match weight of 1. So if the constraint weight is `10hard` it reduces the score by `-10hard`.
```
Comment (briandooley) [inline doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc:7]: I assume "10hard" is correct and not 2 words?

File: doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        constraintVerifier.verifyThat(TimeTableConstraintProvider::roomConflict)
+                .given(firstLesson, conflictingLesson, nonConflictingLesson)
+                .penalizesBy(1);
+    }
+
+}
+----
+
+Notice how `ConstraintVerifier` ignores the constraint weight during testing even if those constraint weights are hard coded in the `ConstraintProvider`. This is because constraint weights change regularly before going into production. This way, constraint weight tweaking does not break the unit tests.
```
Comment (briandooley) [inline doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc]:
```suggestion
Notice how `ConstraintVerifier` ignores the constraint weight during testing even if those constraint weights are hardcoded in the `ConstraintProvider`. This is because constraint weights change regularly before going into production. This way, constraint weight tweaking does not break the unit tests.
```

[Style Guide](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#h) says that "hard code" is one word, at least for the verb. Not sure if if it applies in this case.

File: doc-content/enterprise-only/optimizer/business-optimizer-test-solver-quarkus-proc.adoc [hunk 4]
```diff
@@ -1,8 +1,8 @@
 
-[id='business-optimizer-test-application-proc_{CONTEXT}']
-= Test the application
+[id='business-optimizer-test-solver-proc_{CONTEXT}']
+= Test the school timetable solver
 
-A good application includes test coverage. This example tests the {PLANNER} school timetable project on {QUARKUS}. It uses a JUnit test to generate a test dataset and send it to the `TimeTableController` to solve.
+This example tests the {PLANNER} school timetable project on {QUARKUS}. It uses a JUnit test to generate a test data set and send it to the `TimeTableController` to solve.
```
Comment (briandooley) [inline doc-content/enterprise-only/optimizer/business-optimizer-test-solver-quarkus-proc.adoc:5]:
```suggestion
This example tests the {PLANNER} school timetable project on the {QUARKUS}. It uses a JUnit test to generate a test data set and send it to the `TimeTableController` to solve.
```

File: doc-content/enterprise-only/optimizer/planner-quarkus-database-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+}
+----
++
+This example includes a `TimeTable` instance. However, you can enable multi-tenancy and handle `TimeTable` instances for multiple schools in parallel.
++
+The `getTimeTable()` method returns the latest timetable from the database. It uses the `ScoreManager` method, which is automatically injected, to calculate the score of that timetable and make it available to the UI.
++
+The `solve()` method starts a job to solve the current timetable and stores the time slot and room assignments in the database. It uses the `SolverManager.solveAndListen()` method to listen to intermediate best solutions and update the database accordingly. This enables the UI to show progress while the backend is still solving.
```
Comment (briandooley) [inline doc-content/enterprise-only/optimizer/planner-quarkus-database-proc.adoc]:
```suggestion
The `solve()` method starts a job to solve the current timetable and stores the time slot and room assignments in the database. It uses the `SolverManager.solveAndListen()` method to listen to intermediate best solutions and update the database accordingly. The UI can use this to show progress while the backend is still solving.
```

Remove [anthropomorphism](https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism). Also, do we know the specific part of the UI that will show this? See [user interface](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#u).

**Review by briandooley — APPROVED** [review]
Some quibbles but no deal breakers. Made some suggestions but happy to approve this.

**Review by dupliaka — APPROVED** [review]

File: doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc (hunk 2 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc:7]: Yes, but good to check.

File: doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc (hunk 3 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc]: Thanks!

File: doc-content/enterprise-only/optimizer/business-optimizer-test-solver-quarkus-proc.adoc (hunk 4 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/optimizer/business-optimizer-test-solver-quarkus-proc.adoc:5]: Actually, it just {QUARKUS}  - Red Hat build of Quarkus. No the. that goes for all of the Red Hat build ofs... I know, it seems wrong handed..
